src/litserve/server.py

```diff
@@ -196,28 +196,22 @@ def __init__(
                 device_list = range(devices)
             self.devices = [self.device_identifiers(accelerator, device) for device in device_list]
 
-        self.workers = self.devices * self.workers_per_device
+        self.inference_workers = self.devices * self.workers_per_device
         self.setup_server()
 
     def launch_inference_worker(self, num_uvicorn_servers: int):
         manager = mp.Manager()
         self.workers_setup_status = manager.dict()
         self.request_queue = manager.Queue()
 
-        self.response_queues = []
-        for _ in range(num_uvicorn_servers):
-            response_queue = manager.Queue()
-            self.response_queues.append(response_queue)
+        self.response_queues = [manager.Queue() for _ in range(num_uvicorn_servers)]
 
         for spec in self._specs:
             # Objects of Server class are referenced (not copied)
             logging.debug(f"shallow copy for Server is created for for spec {spec}")
             server_copy = copy.copy(self)
             del server_copy.app
-            try:
-                spec.setup(server_copy)
-            except Exception as e:
-                raise e
+            spec.setup(server_copy)
 
         process_list = []
         for worker_id, device in enumerate(self.devices * self.workers_per_device):
```

Comment:
Can we use `self.inference_workers` instead of `self.devices * self.workers_per_device` since they refer to the same calculation?

Reply:
great catch! updated.